### PR TITLE
Assign EP Button

### PR DESCRIPTION
**Pull request by mdbenjam**
Update the assign EP page so that when you click the Assign EP button the database is updated to link to the EP. The manager and worker views are also updated to show the new completion status.

Connects: #621 

**Test Plan**
- [x] Manually
- [x] Write new automated tests

cc: @joofsh @sharonwarner

### CODE REVIEW

File: app/models/task.rb [hunk 1]
```diff
@@ -118,7 +123,7 @@ def assigned?
 
   def progress_status
     if completed_at
-      completion_status_text
+      "Completed"
```
Comment (mdbenjam) [inline app/models/task.rb]: This way the manager page will on show the status, not the specific completion code.

File: app/services/external_api/bgs_service.rb [hunk 2]
```diff
@@ -18,7 +18,7 @@ def init_client
     BGS::Services.new(
       env: Rails.application.config.bgs_environment,
       application: "CASEFLOW",
-      client_ip: request.remote_ip,
+      client_ip: current_user.ip_address,
```
Comment (mdbenjam) [inline app/services/external_api/bgs_service.rb]: Looks like this was wrong in the last PR. Fixing it here.

File: app/models/task.rb [hunk 3]
```diff
@@ -92,6 +93,10 @@ def expire!
     complete_and_recreate!(:expired)
   end
 
+  def assign_existing_ep!(end_product_id)
```
Comment (joofsh) [inline app/models/task.rb]: `assign_existing_end_product`

File: app/models/task.rb [hunk 4]
```diff
@@ -137,17 +142,22 @@ def canceled?
   end
 
   # completion_status is 0 for success, or non-zero to specify another completed case
-  def complete!(status)
+  def complete!(status, outgoing_status = nil)
```
Comment (joofsh) [inline app/models/task.rb]: gah we're overlapping again.  Two things:
- The parameter name should match the attribute it updates. `outgoing_reference_id`. 
- We should start using keyword arguments for clarity. So `status:, outgoing_reference_id:`. `complete!(0)` is not super clear imo, better as `complete!(status: 0)`. I've already done this in my PR but looks like you'll beat me in again so this means changing all the references to `complete!` like https://github.com/department-of-veterans-affairs/caseflow/pull/648/files#diff-c95c9451cf097bdd58f73d0f9b114f97R87

File: app/models/task.rb [hunk 5]
```diff
@@ -137,17 +142,22 @@ def canceled?
...
-      completion_status: status
+      completion_status: status,
+      outgoing_reference_id: outgoing_status
     )
   end
 
   def completion_status_text
-    COMPLETION_STATUS_MAPPING.key(completion_status).to_s.titleize
+    if self.class.completion_status_code(:assigned_existing_ep) == completion_status
+      "Assigned Existing EP"
```
Comment (joofsh) [inline app/models/task.rb]: why is this needed uniquely for this one completion code? Does it not titlelize properly?

File: client/app/containers/EstablishClaimPage/EstablishClaimAssociateEP.jsx [hunk 6]
```diff
@@ -13,11 +14,43 @@ export default class AssociatePage extends React.Component {
...
   ];
 
+  handleAssignEndProduct = (endProduct) => () => {
+    let { id } = this.props.task;
+    let { handleAlert, handleAlertClear } = this.props;
+
+    event.preventDefault();
+    handleAlertClear();
+
+    this.setState({
```
Comment (joofsh) [inline client/app/containers/EstablishClaimPage/EstablishClaimAssociateEP.jsx]: if we're using a state, we should create a constructor at the top of this component to initially define this value

File: app/controllers/establish_claims_controller.rb [hunk 7]
```diff
@@ -7,6 +7,11 @@ def perform
     render json: {}
   end
 
+  def select_ep
```
Comment (joofsh) [inline app/controllers/establish_claims_controller.rb]: Can we align the route name closer to what it's doing. You use "select" in the route but the method it calls is "assign". 

 `ep` => `end-product` for clarity

File: app/models/task.rb (hunk 5 above)
Comment (mdbenjam) [inline app/models/task.rb]: Because it doesn't capitalize the p in EP :(. Definitely open to other suggestions. I hated writing this.

Comment (joofsh) [inline app/models/task.rb]: ahh haha, that sucks.

What about a `COMPLETION_STATUS_TEXT_MAPPING =  {  assigned_existing_ep: "Assigned Existing EP" }`. If it finds a mapping in that constant, it will use it, otherwise it will titlelize the symbol version like we use in most cases

File: app/services/external_api/bgs_service.rb (hunk 2 above)
Comment (joofsh) [inline app/services/external_api/bgs_service.rb]: Nice, good catch 👍

File: client/app/containers/EstablishClaimPage/EstablishClaimAssociateEP.jsx [hunk 8]
```diff
@@ -18,6 +26,7 @@ export default class AssociatePage extends React.Component {
         name="Assign to Claim"
         classNames={["usa-button-outline"]}
         onClick={this.handleAssignEndProduct(endProduct)}
+        loading={this.state.loading==endProduct.benefit_claim_id}
```
Comment (joofsh) [inline client/app/containers/EstablishClaimPage/EstablishClaimAssociateEP.jsx]: Nice! small nit but do you mind adding spaces around `===` for readability please

File: app/models/task.rb (hunk 1 above)
Comment (joofsh) [inline app/models/task.rb]: I'm not familiar with the product requirements here. Does the manager versus worker index page require different status text?

Comment (mdbenjam) [inline app/models/task.rb]: Yeah "Completed" in the manager view vs "Assigned to an EP" in the user view.
